airbyte-integrations/connectors/source-webflow/source_webflow/source.py

```diff
@@ -0,0 +1,363 @@
+        # The items API returns records inside a container list called "items"
+        for item in response_json["items"]:
+            yield item
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """
+        Webflow has an API,but it is not consistent with json-schema.org schemas. We use the Schema stream
+        to get these schemas and to also map them to json-schema format.
+        """
+
+        collection_id = self.collection_id
+        schema_stream = Schema(authenticator=self.authenticator, collection_id=collection_id)
+        schema_records = schema_stream.read_records(sync_mode="full_refresh")
+
+        # each record corresponds to a property in the json schema. So we loop over each of these properties
+        # and add it to the json schema.
+        json_schema = {}
+        for schema_property in schema_records:
+            json_schema.update(schema_property)
+
+        return {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "additionalProperties": True,
+            "type": "object",
+            "properties": json_schema,
+        }
+
+
+class SourceWebflow(AbstractSource):
+
+    """This is the main class that defines the methods that will be called by Airbyte infrastructure"""
+
+    def get_collection_name_to_id_dict(self, authenticator: str = None, site_id: str = None) -> Mapping[str, str]:
+        """
+        Most of the Webflow APIs require the collection id, but the streams that we are generating use the collection name.
+        This function will return a dictionary containing collection_name: collection_id entries.
+        """
+
+        collection_name_to_id_dict = {}
+
+        collections_stream = Collections(authenticator=authenticator, site_id=site_id)
+        collections_records = collections_stream.read_records(sync_mode="full_refresh")
+
+        # Loop over the list of records and create a dictionary with name as key, and _id as value
+        for collection_obj in collections_records:
+            collection_name_to_id_dict[collection_obj["name"]] = collection_obj["_id"]
+
+        return collection_name_to_id_dict
+
+    @staticmethod
+    def get_authenticator(config):
+        """
+        Verifies that the information for setting the header has been set, and returns a class
+        which overloads that standard authentication to include additional headers that are required by Webflow.
+        """
+        api_key = config.get("api_key", None)
+        accept_version = config.get("accept_version", None)
+        if not api_key:
+            raise Exception("Config validation error: 'api_key' is a required property")
+        if not accept_version:
+            raise Exception("Config validation error: 'accept_version' is a required property")
+        return WebflowTokenAuthenticator(token=api_key, accept_version=accept_version)
+
+    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
```

Comment:
@sajarin can you check other connectors how they are doing the check connection step?

Reply:
I re-write the connection check to hit an API from which we are pulling records